presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -176,7 +177,8 @@ public void start(HiveSplitSource splitSource)
     {
         this.hiveSplitSource = splitSource;
         for (int i = 0; i < loaderConcurrency; i++) {
-            ResumableTasks.submit(executor, new HiveSplitLoaderTask());
+            ListenableFuture<?> future = ResumableTasks.submit(executor, new HiveSplitLoaderTask());
+            addExceptionCallback(future, hiveSplitSource::fail); // best effort; hiveSplitSource could be already completed
```

Comment:
How it could complete without setting `completionFuture` in `ResumableTasks.java `?

Reply:
explained offline